firebase/FirestoreCtrl.ts

```diff
@@ -116,11 +117,40 @@ export default class FirestoreCtrl {
     }
   }
 
+  /**
+   * Get the name of a user by their UID.
+   */
   async getName(id: string) {
     const user = await this.getUser(id);
     return user?.name;
   }
 
+  /**
+   * Set the name of a user by their UID.
+   */
+  async setName(id: string, name: string) {
+    const user = await this.getUser(id);
+    user.name = name;
+    await this.createUser(id, user);
```

Comment:
I don't think it is good practise to createUser everytime we make a change. Maybe you could use the function updateDoc() from `firebase/firestore`. I dont' know if it works, but you could try something like this : 
`async setName(id: string, name: string) {
  const userRef = doc(firestore, "users", id);
  await updateDoc(userRef, { name });
}`

Reply:
I used createUser because the doc says it should be used for creating and updating a user, I also think it's the most straightforward way of doing it as it's a little set of vars in DBUser.